tableauserverclient/server/server.py

```diff
@@ -1,8 +1,47 @@
 from .exceptions import NotSignedInError
 from .endpoint import Sites, Views, Users, Groups, Workbooks, Datasources, Projects, Auth, Schedules, ServerInfo
+from . import RequestOptions
+
 import requests
 
 
+class Pager(object):
+    """ This class returns a generator that will iterate over all of the results.
+
+    server is the server object that will be used when calling the callback.  It will be passed
+    to the callback on each iteration
+
+    Callback is expected to take a server object and a request options and return two values, an array of results,
+    and the pagination item from the current call.  This will be used to build subsequent requests.
+    """
+
+    def __init__(self, fetcher, opts=None):
+        self._fetcher = fetcher.get
+        self._options = opts
+
+    def __call__(self):
+        current_item_list, last_pagination_item = self._fetcher(self._options)
```

Comment:
this won't work if they set a page number in the options because count will never be greater than total available until we hit the end and throw an error.

We should check that a page number hasn't been set in the options, and if it has output a warning or throw an exception.